extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/S3UploadManager.java

```diff
@@ -45,17 +49,28 @@
 public class S3UploadManager
 {
   private final ExecutorService uploadExecutor;
+  private final ServiceEmitter emitter;
 
   private static final Logger log = new Logger(S3UploadManager.class);
 
+  // For metrics regarding uploadExecutor.
+  private final AtomicInteger queueSize = new AtomicInteger(0);
+
+  // Metric related constants.
+  private static final String METRIC_PREFIX = "s3upload/threadPool/";
+  private static final String TASK_QUEUED_DURATION_METRIC = METRIC_PREFIX + "taskQueuedDuration";
+  private static final String NUM_TASKS_QUEUED_METRIC = METRIC_PREFIX + "queuedTasks";
+  private static final String TASK_DURATION_METRIC = METRIC_PREFIX + "taskDuration";
```

Comment:
Nit: style suggestion

By convention, most devs keep the constants at the top of the class and then all the member fields together.